[BUG] NPE in SimpleACLPermission leads to server errors (status code 500)

**Describe the bug**

It is not possible to use most of the built in applications (eXide, dashboard, ...) with the current develop branch (commit 302e6b5b5565922176b5897149203d944a5c6a76).

As already mentioned here: https://github.com/eXist-db/dashboard/issues/199

The reason is an NPE that is thrown in SimpleACLPermission.validate because the user parameter is null.
I was able to trace it to a call to DBBroker.getCurrentSubject that returns null in DBSource.validate(mode).

https://github.com/eXist-db/exist/blob/develop/exist-core/src/main/java/org/exist/source/DBSource.java#L174

**Expected behavior**

Built in applications to work with the current develop HEAD.

**To Reproduce**

1. Build exist from source (tested with 302e6b5b5565922176b5897149203d944a5c6a76)
2. start exist
3. open dashboard

**Context (please always complete the following information):**
 - eXist-db version: 5.4.0-SNAPSHOT 302e6b5b5565922176b5897149203d944a5c6a76
 - Java Version Adopt OpenJDK 8

**Additional context**
- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`? none

Confirmed :-(

```
HTTP ERROR 500 javax.servlet.ServletException: javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
URI:	/exist/apps/dashboard/
STATUS:	500
MESSAGE:	javax.servlet.ServletException: javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
SERVLET:	XQueryURLRewrite
CAUSED BY:	javax.servlet.ServletException: javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
CAUSED BY:	javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
CAUSED BY:	java.lang.NullPointerException
Caused by:

javax.servlet.ServletException: javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:162)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:763)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:386)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.servlet.ServletException: An error occurred while processing request to /exist/apps/dashboard/: null
	at org.exist.http.urlrewrite.XQueryURLRewrite.service(XQueryURLRewrite.java:366)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1450)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:228)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1434)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1349)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:234)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
	... 18 more
Caused by: java.lang.NullPointerException
	at org.exist.security.SimpleACLPermission.validate(SimpleACLPermission.java:317)
	at org.exist.source.DBSource.validate(DBSource.java:190)
	at org.exist.source.DBSource.validate(DBSource.java:174)
	at org.exist.xquery.XQuery.execute(XQuery.java:311)
	at org.exist.xquery.XQuery.execute(XQuery.java:295)
	at org.exist.http.urlrewrite.XQueryURLRewrite.runQuery(XQueryURLRewrite.java:675)
	at org.exist.http.urlrewrite.XQueryURLRewrite.service(XQueryURLRewrite.java:239)
	... 41 more

Powered by Jetty:// 9.4.43.v20210629
```

--- a/exist-core/src/main/java/org/exist/source/DBSource.java
+++ b/exist-core/src/main/java/org/exist/source/DBSource.java
@@ -171,7 +171,10 @@ public class DBSource extends AbstractSource {
     @Deprecated
     public void validate(final int mode) throws PermissionDeniedException {
         //TODO(AR) This check should not even be here! Its up to the database to refuse access not requesting source
-        validate(broker.getCurrentSubject(), mode);
+        final Subject subject = broker.getCurrentSubject();
+        if (subject != null) {
+            doValidation(subject, mode);
+        }
     }
 
     /**
@@ -187,12 +190,21 @@ public class DBSource extends AbstractSource {
     @Deprecated
     public void validate(final Subject subject, final int mode) throws PermissionDeniedException {
         //TODO(AR) This check should not even be here! Its up to the database to refuse access not requesting source
+        if (subject == null) {
+            final String modeStr = new UnixStylePermissionAider(mode).toString();
+            throw new PermissionDeniedException("Subject not given for checking  '" + modeStr + "' access to resource '" + doc.getURI() + "'.");
+        } else {
+            doValidation(subject, mode);
+        }
+    }
+
+    private void doValidation(final Subject subject, final int mode) throws PermissionDeniedException {
         if (!doc.getPermissions().validate(subject, mode)) {
             final String modeStr = new UnixStylePermissionAider(mode).toString();
             throw new PermissionDeniedException("Subject '" + subject.getName() + "' does not have '" + modeStr + "' access to resource '" + doc.getURI() + "'.");
         }
     }
-    
+
     public Permission getPermissions() {
         return doc.getPermissions();
     }

